src/Microsoft.Azure.CosmosEventSourcing/Events/AtomicEvent.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Azure.CosmosEventSourcing.Aggregates;
+using Microsoft.Azure.CosmosEventSourcing.Stores;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.CosmosEventSourcing.Events;
+
+/// <summary>
+/// An event that is used to pass between <see cref="IAggregateRoot"/>'s and the <see cref="IEventStore{TEventItem}"/>
+/// <remarks>This is used to provide optimistic concurrency control for batches of <see cref="DomainEvent"/>'s.</remarks>
+/// </summary>
+public record AtomicEvent : DomainEvent
+{
+    [JsonConstructor]
+    internal AtomicEvent(Guid id, string eTag)
+    {
+        Id = id;
+        ETag = eTag;
+    }
+
+    internal Guid Id { get; set; }
+
+    internal string ETag { get; set; }
```

Comment:
```suggestion
    internal string ETag { get; init; }
```

Reply:
Ah not sure it liked this, I think I tried it.

Reply:
Ah, are you assigning to this from anything other than the `.ctor` or the object initializer? If not, then this should work.

Reply:
Ah yes, I do update this in the `AggregateRoot`

Reply:
![Screenshot 2022-03-10 at 14 14 25](https://user-images.githubusercontent.com/23740684/157679823-f326f807-ad9b-4c55-afdd-365e313ed279.png)
Ah yes this was it!

Reply:
Is this just a NuGet package I need to add?

Reply:
Ah, ok. I see the problem. This is a project that targets `netstandard2.1`. There is a common `Compat.cs` class that adds the markers for this, see:

https://github.com/IEvangelist/blazorators/blob/main/src/Blazor.SourceGenerators/Compat.cs

```csharp
using System.ComponentModel;

namespace System.Runtime.CompilerServices;

[EditorBrowsable(EditorBrowsableState.Never)]
internal class IsExternalInit { }
```

Also see this: https://stackoverflow.com/a/62656145/2410379

If you add the _Compat.cs_ file, it will compile but more importantly at runtime the actual record `init` bits will function as expected.